src/main/java/edu/hw1/Task3.java

```diff
@@ -7,6 +7,9 @@ private Task3() {
     }
 
     public static boolean isNestable(int[] a1, int[] a2) {
+        if ((a1.length == 0) | (a2.length == 0)) {
```

Comment:
`BUG` давай всё-таки делать логическим ленивым оператором 
```suggestion
        if ((a1.length == 0) || (a2.length == 0)) {
```